packages/core/package-manager/src/installPackage.js

```diff
@@ -128,13 +129,19 @@ async function determinePackageInstaller(
   let configFile = await resolveConfig(fs, filepath, [
     'yarn.lock',
     'package-lock.json',
+    'pnpm-lock.yaml',
   ]);
   let hasYarn = await Yarn.exists();
 
   // If Yarn isn't available, or there is a package-lock.json file, use npm.
   let configName = configFile && path.basename(configFile);
-  if (!hasYarn || configName === 'package-lock.json') {
-    return new Npm();
+  if (!hasYarn || configName) {
```

Comment:
If the Yarn binary doesn't exist and no lockfile is present, NPM should be used by default. However, if there is no lockfile, this new logic will cause Yarn to be used even when the binary doesn't exist.

Reply:
> could you add some integration tests? You can take inspiration from how we did it for yarn 2



> If the Yarn binary doesn't exist and no lockfile is present, NPM should be used by default. However, if there is no lockfile, this new logic will cause Yarn to be used even when the binary doesn't exist.

Yes, I made a mistake.